GitUI/Avatars/InitialsAvatarProvider.cs

```diff
@@ -16,27 +16,49 @@ public class InitialsAvatarProvider : IAvatarProvider
         /// <inheritdoc/>
         public Task<Image?> GetAvatarAsync(string email, string? name, int imageSize)
         {
-            var (initials, hashCode) = GetInitialsAndHashCode(email, name);
+            (string initials, int hashCode) = GetInitialsAndHashCode(email, name);
 
-            var avatarColor = _avatarColors[Math.Abs(hashCode) % _avatarColors.Length];
-            var avatar = DrawText(initials, avatarColor, imageSize);
+            Color avatarColor = _avatarColors[hashCode];
+            Image avatar = DrawText(initials, avatarColor, imageSize);
 
             return Task.FromResult<Image?>(avatar);
         }
 
+        /// <summary>
+        /// Calculate the most simpler non-cryptographic deterministic hash (to get the same result every times it is calculated for the same string)
+        /// We just need a not too costly way to convert a string in an integer and calculating a hash is a good way to do it.
+        /// We are avoiding to use `.GetHashCode()` because it returns a different value for each process :(
+        /// Taken from https://stackoverflow.com/a/5155015
+        /// </summary>
+        /// <param name="str">the string to calculate the hash for</param>
+        /// <returns>the hash calculated</returns>
```

Comment:
```suggestion
        /// <summary>
        /// Calculate the most simpler non-cryptographic deterministic hash (to get the same result every times it is calculated for the same string)
        /// We just need an inexpensive way to convert a string to an integer, and calculating a hash is a good way to do it.
        /// We are not using <c>GetHashCode()</c> because it returns a different value for each process.
        /// Borrowed from https://stackoverflow.com/a/5155015
        /// </summary>
        /// <param name="str">The string to calculate a hash for.</param>
        /// <returns>The calculated hash.</returns>
```